modules/javafx.graphics/src/main/java/javafx/scene/robot/Robot.java

```diff
@@ -0,0 +1,373 @@
+        Objects.requireNonNull(location);
+        mouseMove(location.getX(), location.getY());
+    }
+
+    /**
+     * Presses the specified {@link MouseButton}s.
+     *
+     * @param buttons the mouse buttons to press
+     * @throws IllegalStateException if this method is called on a thread
+     * other than the JavaFX Application Thread.
+     */
+    public void mousePress(MouseButton... buttons) {
+        peer.mousePress(buttons);
+    }
+
+    /**
+     * Releases the specified {@link MouseButton}s.
+     *
+     * @param buttons the mouse buttons to release
+     * @throws IllegalStateException if this method is called on a thread
+     * other than the JavaFX Application Thread.
+     */
+    public void mouseRelease(MouseButton... buttons) {
+        peer.mouseRelease(buttons);
+    }
+
+    /**
+     * Clicks the specified {@link MouseButton}s.
+     *
+     * @implSpec This is a convenience method that is equivalent to calling
+     * {@link #mousePress(MouseButton...)} followed by {@link #mouseRelease(MouseButton...)}.
+     * @param buttons the mouse buttons to click
+     * @throws IllegalStateException if this method is called on a thread
+     * other than the JavaFX Application Thread.
+     */
+    public void mouseClick(MouseButton... buttons) {
+        mousePress(buttons);
+        mouseRelease(buttons);
+    }
+
+    /**
+     * Scrolls the mouse wheel by the specified amount. Positive {@code wheelAmt}
+     * scrolls up whereas negative scrolls down.
+     *
+     * @param wheelAmt the (signed) amount to scroll the wheel
+     * @throws IllegalStateException if this method is called on a thread
+     * other than the JavaFX Application Thread.
+     */
+    public void mouseWheel(int wheelAmt) {
+        peer.mouseWheel(wheelAmt);
+    }
+
+    /**
+     * Returns the {@link Color} of the pixel at the specified screen coordinates of the
+     * primary screen.
+     *
+     * @param x the x coordinate to get the pixel color from
+     * @param y the y coordinate to get the pixel color from
+     * @return the pixel color at the specified screen coordinates
+     * @throws IllegalStateException if this method is called on a thread
+     * other than the JavaFX Application Thread.
+     */
+    public Color getPixelColor(double x, double y) {
+        return peer.getPixelColor(x, y);
+    }
+
+    /**
+     * Returns the {@link Color} of the pixel at the screen coordinates relative to the
+     * primary screen specified by {@code location}.
+     *
+     * @param location the (x,y) coordinates to get the pixel color from
+     * @return the pixel color at the specified screen coordinates
+     * @throws NullPointerException if the given {@code location} is {@literal null}
+     * @throws IllegalStateException if this method is called on a thread
+     * other than the JavaFX Application Thread.
+     */
+    public Color getPixelColor(Point2D location) {
+        Objects.requireNonNull(location);
```

Comment:
null check w/o additional info is not necessary, npe will be thrown by the next line anyway.